src/ds5/ds5-device.cpp

```diff
@@ -667,6 +667,7 @@ namespace librealsense
 
         depth_ep->register_option(RS2_OPTION_GLOBAL_TIME_ENABLED, enable_global_time_option);
 
+        depth_ep->register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_Y8I, RS2_STREAM_INFRARED));
```

Comment:
@puzzlepaint , thank you for the suggestion.
At this point the requirement are to expose Left/Right Monochrome intensity streams only.
And while the camera streams it as interleaved `Y8I`, the users of the API would continue to receive it as separate frames.
Internally  though, we do utilize Y8I for record and playback capability.

The de-interleaving routine is shouldn't be CPU-intensive even for low-end boards. Can you provide some details on the exact use-case and explain the application need for exposing the raw data format?

Reply:
Thanks for your reply. My current use-case is recording the video of a D415 camera (color stream at 1920 x 1080, 30 Hz; both infrared streams at 1280 x 720, 30 Hz) on a Raspberry PI 4B (2 GB) to an SSD attached via USB. I implemented my own recording code such that I have control over the file format and can directly write some additional information into it (such as frame timestamps that are synchronized with other devices over a network).

Writing the video data to the SSD seems to cause significant CPU load according to `htop`, even after switching from `fwrite()` to the `write()` syscall. Because of this, there is unfortunately a risk of frame drops. Initially, I tried to use the V4L2 backend, but it dropped many frames during recording. Implementing some optimizations (such as #9139, or this PR) reduced the amount of dropped frames, but did not eliminate frame drops. I then switched to the RSUSB backend which generally seemed to work better, but I also observed some (rare) frame drops with it.

So, the need for optimizations such as using the Y8I format comes from the risk of frame drops in the presence of high CPU load (e.g., due to recording as described above) and possibly load spikes (e.g., due to background processes waking up). For this, I think that it makes sense to avoid any unnecessary data processing.

Regarding CPU/memory-intensiveness, the de-interleaving step can possibly be compared to an "unoptimized memcpy of the whole frame with worse memory locality":
* "unoptimized", because it does not make use of platform-specific intrinsics (e.g., armv8 intrinsics on the Raspberry Pi 4), unless the compiler happens to be able to auto-vectorize the code.
* "worse memory locality", because the data is copied to two separate buffers instead of to a single location as in standard memcpy.

So, yes, under usual conditions the Raspberry PI 4 is more than fast enough to do processing steps like this, but it is the load spikes that I am concerned about, since I sometimes observe frame drops whose occurrence I would like to minimize.

Reply:
@puzzlepaint , I'll bring the request for internal review and update accordingly